### PR TITLE
Members update

### PR DESCRIPTION
**Pull request by Ahmedgagan**
`counting deleted topics in assignment_counts` issue solved

### CODE REVIEW

File: app/controllers/discourse_assign/assign_controller.rb [hunk 1]
```diff
@@ -137,8 +137,8 @@ def group_members
       guardian.ensure_can_see_group_members!(group)
 
       members = User
-        .joins("LEFT OUTER JOIN group_users g on users.id=g.user_id LEFT OUTER JOIN user_options uo on uo.user_id=users.id LEFT OUTER JOIN topic_custom_fields tcf ON tcf.value::int = users.id")
-        .where("tcf.name = 'assigned_to_id' AND g.group_id=? AND (users.id > 0)", group.id)
+        .joins("LEFT OUTER JOIN group_users g on users.id=g.user_id LEFT OUTER JOIN user_options uo on uo.user_id=users.id LEFT OUTER JOIN topic_custom_fields tcf ON tcf.value::int = users.id LEFT OUTER JOIN topics t ON t.id = tcf.topic_id")
+        .where("tcf.name = 'assigned_to_id' AND g.group_id=? AND (users.id > 0) AND t.deleted_by_id IS NULL", group.id)
```
Comment (davidtaylorhq) [inline app/controllers/discourse_assign/assign_controller.rb:141]: Why use deleted_by_id here, and deleted_at later? Can we switch them both to deleted_at

Comment (Ahmedgagan) [reply app/controllers/discourse_assign/assign_controller.rb:141]: yaa sure, might be some misunderstanding😅

File: app/controllers/discourse_assign/assign_controller.rb [hunk 2]
```diff
@@ -137,8 +137,8 @@ def group_members
       guardian.ensure_can_see_group_members!(group)
 
       members = User
-        .joins("LEFT OUTER JOIN group_users g on users.id=g.user_id LEFT OUTER JOIN user_options uo on uo.user_id=users.id LEFT OUTER JOIN topic_custom_fields tcf ON tcf.value::int = users.id")
-        .where("tcf.name = 'assigned_to_id' AND g.group_id=? AND (users.id > 0)", group.id)
+        .joins("LEFT OUTER JOIN group_users g on users.id=g.user_id LEFT OUTER JOIN user_options uo on uo.user_id=users.id LEFT OUTER JOIN topic_custom_fields tcf ON tcf.value::int = users.id LEFT OUTER JOIN topics t ON t.id = tcf.topic_id")
```
Comment (davidtaylorhq) [inline app/controllers/discourse_assign/assign_controller.rb:140]: This is getting a little long, can we split it into multiple `joins` statements?

**Comment by davidtaylorhq** [comment]
Would be good to add a ruby test for this as well to check that deleted topics are not included in either count

File: app/controllers/discourse_assign/assign_controller.rb [hunk 3]
```diff
@@ -137,8 +137,11 @@ def group_members
       guardian.ensure_can_see_group_members!(group)
 
       members = User
-        .joins("LEFT OUTER JOIN group_users g on users.id=g.user_id LEFT OUTER JOIN user_options uo on uo.user_id=users.id LEFT OUTER JOIN topic_custom_fields tcf ON tcf.value::int = users.id")
-        .where("tcf.name = 'assigned_to_id' AND g.group_id=? AND (users.id > 0)", group.id)
+        .joins("LEFT OUTER JOIN group_users g on users.id=g.user_id")
+        .joins("LEFT OUTER JOIN user_options uo on uo.user_id=users.id")
```
Comment (davidtaylorhq) [inline app/controllers/discourse_assign/assign_controller.rb:141]: Are we using user_options for anything?

File: plugin.rb [hunk 4]
```diff
@@ -42,7 +42,7 @@
 
   add_to_serializer(:group_show, :assignment_count) do
     Topic.joins("JOIN topic_custom_fields tcf ON topics.id = tcf.topic_id AND tcf.name = 'assigned_to_id' AND tcf.value IS NOT NULL")
-      .where("tcf.value IN (SELECT group_users.user_id::varchar(255) FROM group_users WHERE (group_id IN (SELECT id FROM groups WHERE name = ?)))", object.name).count
+      .where("tcf.value IN (SELECT group_users.user_id::varchar(255) FROM group_users WHERE (group_id IN (SELECT id FROM groups WHERE name = ?))) AND topics.deleted_at IS NULL", object.name).count
```
Comment (davidtaylorhq) [inline plugin.rb:45]: Can we split this into multiple `where` calls as well? It is pretty hard to read at the moment

File: plugin.rb [hunk 5]
```diff
@@ -42,7 +42,9 @@
 
   add_to_serializer(:group_show, :assignment_count) do
     Topic.joins("JOIN topic_custom_fields tcf ON topics.id = tcf.topic_id AND tcf.name = 'assigned_to_id' AND tcf.value IS NOT NULL")
-      .where("tcf.value IN (SELECT group_users.user_id::varchar(255) FROM group_users WHERE (group_id IN (SELECT id FROM groups WHERE name = ?)))", object.name).count
+      .where("tcf.value IN (SELECT group_users.user_id::varchar(255) FROM group_users WHERE (group_id IN (SELECT id FROM groups WHERE name = ?)))", object.name)
+      .where("AND topics.deleted_at IS NULL")
```
Comment (davidtaylorhq) [inline plugin.rb:46]:
```suggestion
      .where("topics.deleted_at IS NULL")
```

**Pull request merged**